src/pe/certificate_table.rs

```diff
@@ -0,0 +1,146 @@
+use crate::error;
+use scroll::Pread;
+
+use alloc::string::ToString;
+use alloc::vec::Vec;
+
+#[repr(u16)]
+#[derive(Debug, PartialEq, Copy, Clone)]
+enum CertificateAttributeRevision {
+    Revision1_0 = 0x0100,
+    Revision2_0 = 0x0200,
+}
+
+impl TryFrom<u16> for CertificateAttributeRevision {
+    type Error = error::Error;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        Ok(match value {
+            x if x == CertificateAttributeRevision::Revision1_0 as u16 => {
+                CertificateAttributeRevision::Revision1_0
+            }
+            x if x == CertificateAttributeRevision::Revision2_0 as u16 => {
+                CertificateAttributeRevision::Revision2_0
+            }
+            _ => {
+                return Err(error::Error::Malformed(
+                    "Invalid certificate attribute revision".to_string(),
+                ))
+            }
+        })
+    }
+}
+
+#[repr(u16)]
+#[derive(Debug)]
+enum CertificateAttributeType {
+    X509 = 0x0001,
+    PkcsSignedData = 0x0002,
+    Reserved1 = 0x0003,
+    TsStackSigned = 0x0004,
+}
+
+impl TryFrom<u16> for CertificateAttributeType {
+    type Error = error::Error;
+
+    fn try_from(value: u16) -> Result<Self, Self::Error> {
+        Ok(match value {
+            x if x == CertificateAttributeType::X509 as u16 => CertificateAttributeType::X509,
+            x if x == CertificateAttributeType::PkcsSignedData as u16 => {
+                CertificateAttributeType::PkcsSignedData
+            }
+            x if x == CertificateAttributeType::Reserved1 as u16 => {
+                CertificateAttributeType::Reserved1
+            }
+            x if x == CertificateAttributeType::TsStackSigned as u16 => {
+                CertificateAttributeType::TsStackSigned
+            }
+            _ => {
+                return Err(error::Error::Malformed(
+                    "Invalid certificate attribute type".to_string(),
+                ))
+            }
+        })
+    }
+}
+
+#[derive(Clone, Pread)]
+pub struct CertificateAttributeHeader {
+    length: u32,
+    revision: u16,
+    certificate_type: u16,
+}
+
+pub const CERTIFICATE_DATA_OFFSET: u32 = 8;
+#[derive(Debug)]
+#[allow(dead_code)]
```

Comment:
why dead code?

Reply:
I had rust-analyzer errors saying that the fields were never read (which is true, but also, because this is a library), maybe this is not the good way to silence them?

Reply:
s/errors/warnings/

Reply:
there should not be any deadcode; if the fields are never read it's because something is not marked pub and nothing can ever read them.